docs/gangplank/usage.md

```diff
@@ -1,3 +1,239 @@
+
+- id: the name of the stage; must be unique
+- command: a list of commands to execute
+- concurrent_execution: bool to indicate if the `commands` can be executed concurrently
+- build_artifacts: known artifacts to build
+- direct_execution: do not apply templating
+- prep_commands: a list of commands to run before command
+- post_commands: a list of commands to run last (such as test or cleanup)
+- post_always: a bool that indicates whether the `post_commands` should _always_ be executed regardless of the success of the `commands` stage.
+- require_artifacts: the type of artifact that's required for work (i.e. `qemu` or `aws`). Stages will not start until the artifact appears.
+
+To illustrate this, consider:
+```
+- id: base Stage
+  build_artifacts:
+  - base
+  post_commands:
+  - cosa kola run --basic-qemu-scenarios
+- id: clouds
+  concurrent_execution: true
+  build_artifacts:
+  - aws
+  - gcp
+  required_artifacts:
+  - qemu
+- id: finalize
+  build_artifacts:
+  - finalize
+```
+
+In this example, Gangplank will:
+
+1. Build and test base artifacts in the first pod
+1. In a new pod, concurrently build AWS and GCP but only after the QEMU artifact is found.
+1. Combine meta.json together
+
+### Meta Data and JobSpec Templating
+
+Gangplank was initially started after belately realizing that the Jenkins Pipelines are, in fact, complicated templating engines. That is, a considerable amount of time, energy and development was put into translating data from YAML/JSON into execution rules.
+
+Gangplank supports rendering commands from the `meta.json` in CoreOS Assembler and the JobSpec via Golang templates. The caveat, however, is that `meta.json` variables appears _after_ the base build. Generally speaking, this means to a base build are defined in the Jobspec while artifacts generated from a base build may use both `meta.json` and the Jobspec. 
+
+#### JobSpec Example
+
+Any JobSpec variable is exposed using the GoLang templating `{{.JobSpec.<VAR>}}`
+
+```
+archives:
+  s3:
+    bucket: darkarts
+    path: magicalmysteries
+stages:
+  - id: prep
+    command:
+    - cosa buildprep s3://{{.JobSpec.Archives.S3.Bucket}}/{{.JobSpec.Archives.S3.Path}}
+```
+
+The above example will run the CoreOS Assembler command `cosa buildprep s3://darkarts/magicalmysteries`. 
+
+#### meta.json
+
+`meta.json` fields become available for any stage that is executed after Gangplank detects a new `meta.json`. Data for a `meta.json` is prefixed using `.Meta`. `meta.json` is always read immediately before a stage is executed (if `meta.json` exists). 
+ 
+```
+stages:
+ - id: build
+   build_artifacts:
+   - base
+ - id: make a file
+   command:
+   - touch {{ .Meta.BuildID }}
+```
+
+#### Complicated rules
+
+With the availability of GoLang templating, the possibility exists to do loops and to dynamically create commands. The following example, publishes an AMI to all AWS regions.
```

Comment:
Idea: Earlier in the document it's denoted that Jenkins boils down to "complicated templating engines." It may be worth adding a cautionary statement here less users continue the pattern into Gangplank.

Reply:
Indeed. Cautionary text added.